build-tools/packages/build-cli/src/commands/typetests.ts

```diff
@@ -163,14 +163,20 @@ function updateTypeTestConfiguration(pkgJson: PackageJson, options: TypeTestConf
 
 		if (!enabled || options.version === VersionOptions.Clear) {
 			// There isn't a type safe alternative to delete here,
-			// and delete should be safe since the key can't collide with anything builtin since it has to end in `-previous` which makes it not a valid identifier.
-			// eslint-disable-next-line @typescript-eslint/no-dynamic-delete
-			delete pkgJson.devDependencies[oldDepName];
+			// and delete should be safe since the key can't collide with anything builtin since it has to end in `-previous`
```

Comment:
I think:
`pkgJson.devDependencies ??= {};`
is a bit more clear then adding these extra cheeks.

But what is more interesting here is that the type is wrong, so the compiler didn't catch it. The PackageJson type incorrectly assumes devDependencies exist and thats the real source of this bug in my opinion.

Either that type should be updated and/or when we convert the parsed data to that type we should check it (for example use a TypeBox schema to validate that it matches the type when doing the conversion so any future case where they mismatch error more clearly).

Reply:
> But what is more interesting here is that the type is wrong, so the compiler didn't catch it. The PackageJson type incorrectly assumes devDependencies exist and thats the real source of this bug in my opinion.

Agreed. That types bug should be fixed in this PR as well.